src/board.py

```diff
@@ -24,14 +25,47 @@ class Direction(IntEnum):
 
 def build():
     fences = [[Direction.NO for i in range(FENCE_SIZE)] for j in range(FENCE_SIZE)]
-    for i in range(10):
-        center_x = randint(0, FENCE_SIZE - 1)
-        center_y = randint(0, FENCE_SIZE - 1)
-        direction = randint(Direction.HORIZONTALLY, Direction.VERTICALLY)
-        fences[center_x][center_y] = direction
+    for i in range(FENCE_NUMBER):
+        added = False
+        while not added:
+            center_x = randint(0, FENCE_SIZE - 1)
+            center_y = randint(0, FENCE_SIZE - 1)
+            direction = randint(Direction.HORIZONTALLY, Direction.VERTICALLY)
+            if can_add_fence(fences, center_x, center_y, direction):
+                fences[center_y][center_x] = direction
+                added = True
     return fences
 
 
+def can_add_fence(fences, x, y, direction):
+    if not are_inside_fences(x, y, fences) or direction == Direction.NO:
```

Comment:
I'm not sure of what test the function `are_inside_fences`.. It test that two fences don't overlap?